cmd/jujud/updateseries/updateseries.go

```diff
@@ -118,38 +139,64 @@ func (c *UpdateSeriesCommand) Run(ctx *cmd.Context) error {
 		return err
 	}
 
+	jujuVersion, err := GetJujuVersion(c.machineAgent, c.dataDir)
+	if err != nil {
+		return err
+	}
+
 	switch {
 	case toInitSys == service.InitSystemUpstart:
 		return errors.NewNotSupported(nil, "downgrade to series using upstart not supported")
 	case toInitSys == fromInitSys && toInitSys == service.InitSystemSystemd:
-		if err = c.copyTools(); err != nil {
+		if err := c.manager.CopyAgentBinary(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries, c.fromSeries, jujuVersion); err != nil {
 			return err
 		} else {
 			ctx.Infof("successfully copied tools and relinked agent tools")
 		}
 		if !c.startAgents {
 			break
 		}
-		if err = c.startAllAgents(ctx); err != nil {
+		if errMachineName, errUnitNames, err := c.manager.StartAllAgents(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries); err != nil {
 			return err
```

Comment:
what if one of the agents suggested in starting, but one failed.  Where does the notice that the one succeeded happen?